### PR TITLE
Added Swift Tests for iOS Image Searcher and include ImageSearcher API in TFLiteTaskVision pod

### PR DESCRIPTION
**Pull request by priankakariatyml**
1. Added Swift Tests
2. Included image searcher in iOS vision framework for coocoapods

### CODE REVIEW

**Comment by priankakariatyml** [comment]
@khanhlvg @schmidt-sebastian Raised a PR for Swift Tests

File: tensorflow_lite_support/ios/BUILD [hunk 1]
```diff
@@ -213,6 +229,7 @@ ios_static_framework(
         ":TFLAudioRecord.h",
         ":TFLAudioTensor.h",
         ":TFLBaseOptions.h",
+        ":TFLExternalFile.h",
```
Comment (khanhlvg) [inline tensorflow_lite_support/ios/BUILD:232]: Why do you need to add TFLExternalFile to the Audio framework?

Comment (priankakariatyml) [reply tensorflow_lite_support/ios/BUILD:232]: [TFLBaseOptions](https://github.com/tensorflow/tflite-support/blob/master/tensorflow_lite_support/ios/task/core/sources/TFLBaseOptions.h#L84) also needs `TFLExternalFile` for defining the model file. Before the image searcher APIs were added `TFLExternalFile` was defined in `TFLBaseOptions.h`.
But searcher needed `TFLExternalFile` in `TFLSearchOptions` as well for defining the index file. Hence the class definition  was moved to a separate file. So all frameworks that use `TFLBaseOptions` needs to include `TFLExternalFile.h` as well.

Comment (khanhlvg) [reply tensorflow_lite_support/ios/BUILD:232]: I see. Thanks for tha clarification.

File: tensorflow_lite_support/ios/test/task/vision/image_searcher/TFLImageSearcherTests.swift [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        mlImage: mlImage))
+    self.validateSearchResult(searchResult)
+  }
+
+  func testCreateImageSearcherWithNoModelPathFails() throws {
+    let options = ImageSearcherOptions()
+    do {
+      let imageSearcher = try ImageSearcher.searcher(
+        options: options)
+      XCTAssertNil(imageSearcher)
```
Comment (khanhlvg) [inline tensorflow_lite_support/ios/test/task/vision/image_searcher/TFLImageSearcherTests.swift:170]: An error is already thrown when you create the imageSearcher so this assert will not get called. You'll need to do the assert outside of the do-catch statement.

I'll fix it when I import the PR.

**Review by khanhlvg — APPROVED** [review]
Just a minor issue that I co

**Pull request merged**